File: Frontend/sma-ui/DTOs/OrderDTOs.ts
export interface OrderItemRequestDto {
  productId: string;
  quantity: number;
}

export interface CreateOrderRequestDto {
  shippingAddress: string;
  items: OrderItemRequestDto[];
}

export interface OrderResponseDto {
  orderId: string;
  totalAmount: number;
  status: string;
  shippingAddress: string;
  createdAt: string;
}

export interface PagedOrderResponseDto {
  items: OrderResponseDto[];
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

export interface AdminOrderResponseDto extends OrderResponseDto {
  userId: string;
  customerEmail: string;
  customerName: string;
}

export interface PagedAdminOrderResponseDto {
  items: AdminOrderResponseDto[];
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

export interface CheckoutResponseDto extends OrderResponseDto {
  checkoutUrl: string;
}

export interface AdminAnalyticsDto {
  grossSales: number;
  paidOrderCount: number;
  orderCount: number;
  averageOrderValue: number;
  inventoryValue: number;
  activeProductCount: number;
  lowStockProductCount: number;
  orderStatusCounts: Record<string, number>;
  dailySales: { date: string; amount: number }[];
}

File: Backend/SMA.API/DTOs/OrderDtos.cs
namespace SMA.API.DTOs
{
    public class OrderItemRequestDto
    {
        public Guid ProductId { get; set; }
        public int Quantity { get; set; }
    }

    public class CreateOrderRequestDto
    {
        public string ShippingAddress { get; set; } = string.Empty;
        public List<OrderItemRequestDto> Items { get; set; } = [];
    }

    // Response response DTO
    public class OrderResponseDto
    {
        public Guid OrderId { get; set; }
        public decimal TotalAmount { get; set; }
        public string Status { get; set; } = string.Empty;
        public string ShippingAddress { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
    }

    public class PagedOrderResponseDto
    {
        public List<OrderResponseDto> Items { get; set; } = [];
        public int Page { get; set; }
        public int PageSize { get; set; }
        public int TotalCount { get; set; }
        public int TotalPages { get; set; }
    }

    public class AdminOrderResponseDto : OrderResponseDto
    {
        public Guid UserId { get; set; }
        public string CustomerEmail { get; set; } = string.Empty;
        public string CustomerName { get; set; } = string.Empty;
    }

    public class PagedAdminOrderResponseDto
    {
        public List<AdminOrderResponseDto> Items { get; set; } = [];
        public int Page { get; set; }
        public int PageSize { get; set; }
        public int TotalCount { get; set; }
        public int TotalPages { get; set; }
    }

    public class CheckoutResponseDto : OrderResponseDto
    {
        public string CheckoutUrl { get; set; } = string.Empty;
    }

    public class AdminAnalyticsDto
    {
        public decimal GrossSales { get; set; }
        public int PaidOrderCount { get; set; }
        public int OrderCount { get; set; }
        public decimal AverageOrderValue { get; set; }
        public decimal InventoryValue { get; set; }
        public int ActiveProductCount { get; set; }
        public int LowStockProductCount { get; set; }
        public Dictionary<string, int> OrderStatusCounts { get; set; } = [];
        public List<DailySalesDto> DailySales { get; set; } = [];
    }

    public class DailySalesDto
    {
        public DateTime Date { get; set; }
        public decimal Amount { get; set; }
    }
}
